eras/babbage/impl/src/Cardano/Ledger/Babbage/Rules/Utxow.hs

```diff
@@ -187,9 +188,11 @@ babbageUtxowTransition = do
   {-  inputHashes  = dom(txdats txw)   -}
   runTest $ missingRequiredDatums utxo tx txbody
 
-  {- inputHashes ⊆ dom(txdats txw) -}
-  -- NEW in BABBAGE
-  runTest $ allInputDataHashesAreKnown inputHashes txw
+  {-
+    {- inputHashes ⊆ dom(txdats txw) -}
+    -- NEW in BABBAGE
+    runTest $ allInputDataHashesAreKnown inputHashes txw
```

Comment:
Was this comment out intentional?

Reply:
Yes there is some question if this rule is superceeded by the shelley rule.

Reply:
Both alonzo and babbage have a check that `inputHashes ⊆ dom(txdats txw)`. this is provided by `missingRequiredDatums` in alonzo, but unfortunately `missingRequiredDatums` also does the reverse containment (or nearly the reverse containment, it also allows for datums to be supplied that correspond to newly created outputs.

So in babbage, we need the part of `missingRequiredDatums` that checks `inputHashes ⊆ dom(txdats txw)`, together with a new version of the reverse containment that also allows for datums attached to reference inputs.

Reply:
OK it has been determined that the previous check, does everything this one does, so the comment out was legitimate, and the comments are now removed.